scripts/chart-templates/crd-assume-ownership/templates/_helpers.tpl

```diff
@@ -5,3 +5,47 @@
 {{- printf "%s/" .Values.global.cattle.systemDefaultRegistry -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Creates a list of ConfigMaps that contain CRD files with the following format.
+It ensures that no ConfigMap is >1mb in size.
+name: $name-manifest-xx
+crds:
+  {{ path }}: {{ File }}
+size: {{ size in characters }}
+*/}}
+{{- define "crd_manifest_configmaps" -}}
+{{- \$currentScope := . -}}
+{{- \$counter := dict "i" 0 -}}
+{{/* Initialize the first configMap and add it to the list of ConfigMaps necessary */}}
+{{- \$currName := (printf "%s-manifest-%02d" \$currentScope.Chart.Name (get \$counter "i")) -}}
+{{- \$currConfigMap := dict "name" \$currName "size" 0 "crds" (dict) -}}
+{{- \$configMaps := dict \$currName \$currConfigMap -}}
+{{/* Iterate through the CRDs to add to the configMap or create a new one */}}
+{{- range \$filepath, \$_ :=  (.Files.Glob "crd-manifest/**.yaml") -}}
+{{- with \$currentScope -}}
+{{/* Get current values from the dictionary */}}
+{{- \$currConfigMap := get \$configMaps \$currName -}}
+{{- \$currSize := get \$currConfigMap "size" -}}
+{{- \$currCRDs := get \$currConfigMap "crds" -}}
+{{/* Get the next CRD file that needs to be added to a ConfigMap */}}
+{{- \$path := base \$filepath -}}
+{{- \$file := .Files.Get \$filepath -}}
+{{/* Check if the file size of the CRD causes size to exceed 1048576 characters, which requires a new ConfigMap */}}
+{{- \$currSize := (add \$currSize (len \$file)) -}}
+{{- if ge \$currSize 1000000 -}}
```

Comment:
This new Helm template function is the crux of how we choose to split the files located in `crd-manifest/` (each file usually represents a CRD) into separate ConfigMaps.

Most of the code here is just verbose due to Helm's constraints on defining templates + the fact that we need to escape `$` characters used by Helm, but I'm hoping the Helm comments might make this a little easier to parse.

In general, this code seeks to:
- Initialize 1 ConfigMap that will be used by CRDs
- Iterate through each CRD to collect information on it (path, file contents, size)
- If there's space in the existing ConfigMap (e.g. file size + current size doesn't exceed `1,000,000`, which is a little less than 1mb), add the CRD to this ConfigMap and continue.
- Otherwise, add an entry for a new ConfigMap and add the CRD to it. Continue to iterate through CRDs with the new ConfigMap until that also fills up.